### PR TITLE
Meta: S3 directory bucket subservice

### PR DESCRIPTION
**Pull request by Laren-AWS**
Add s3-directory-buckets as a pseudo-subservice bundled under s3 so that its examples are published in a separate folder under s3 in the documentation.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws_doc_sdk_examples_tools/config/services.yaml [hunk 1]
```diff
@@ -2773,6 +2773,22 @@ s3-control:
   tags:
     product_categories: {'Storage'}
   version: s3control-2018-08-20
+s3-directory-buckets:
+  bundle: s3
+  api_ref: AmazonS3/latest/API/Welcome.html
+  blurb: are a high-performance, single-zone Amazon S3 storage class that is purpose-built to deliver consistent, single-digit millisecond data access for your most latency-sensitive applications.
```
Comment (yanjieniu) [inline aws_doc_sdk_examples_tools/config/services.yaml:2779]: are designed to store data within a single AWS Zone. Directory buckets organize data hierarchically into directories, providing a structure similar to a file system.

File: aws_doc_sdk_examples_tools/config/services.yaml [hunk 2]
```diff
@@ -2773,6 +2773,22 @@ s3-control:
...
+  expanded:
+    long: Amazon S3 Directory Buckets
+    short: Amazon S3 Directory Buckets
```
Comment (yanjieniu) [inline aws_doc_sdk_examples_tools/config/services.yaml:2782]: S3 Directory Buckets

File: aws_doc_sdk_examples_tools/config/services.yaml [hunk 3]
```diff
@@ -2773,6 +2773,22 @@ s3-control:
...
+  guide:
+    subtitle: User Guide
+    url: AmazonS3/latest/userguide/s3-express-one-zone.html
```
Comment (yanjieniu) [inline aws_doc_sdk_examples_tools/config/services.yaml:2785]: AmazonS3/latest/userguide/directory-buckets-overview.html

File: aws_doc_sdk_examples_tools/config/services.yaml (hunk 1 above)
Comment (yanjieniu) [inline aws_doc_sdk_examples_tools/config/services.yaml:2779]: are designed to store data within a single AWS Zone. Directory buckets organize data hierarchically into directories, providing a structure similar to a file system.

File: aws_doc_sdk_examples_tools/config/services.yaml [hunk 4]
```diff
@@ -2773,6 +2773,22 @@ s3-control:
...
+  long: '&S3; Directory Buckets'
+  short: '&S3; Directory Buckets'
+  sort: S3 Directory Buckets
+  tags:
+    product_categories: {'Storage'}
+  version: s3-2006-03-01
```
Comment (yanjieniu) [inline aws_doc_sdk_examples_tools/config/services.yaml:2791]: s3directorybuckets-2023-11-28

**Review by yanjieniu — CHANGES_REQUESTED** [review]
Update information for the feature of S3 directory buckets.

**Review by yanjieniu — COMMENTED** [review]
SDK config comment based on the "s3-directory-buckets: s3" change.

File: aws_doc_sdk_examples_tools/config/services.yaml [hunk 5]
```diff
@@ -2773,6 +2773,22 @@ s3-control:
...
+  api_ref: AmazonS3/latest/API/Welcome.html
+  blurb: are a high-performance, single-zone Amazon S3 storage class that is purpose-built to deliver consistent, single-digit millisecond data access for your most latency-sensitive applications.
+  expanded:
+    long: Amazon S3 Directory Buckets
+    short: Amazon S3 Directory Buckets
+  guide:
+    subtitle: User Guide
+    url: AmazonS3/latest/userguide/s3-express-one-zone.html
+  long: '&S3; Directory Buckets'
+  short: '&S3; Directory Buckets'
```
Comment (yanjieniu) [inline aws_doc_sdk_examples_tools/config/services.yaml:2787]: S3 Directory Buckets

File: aws_doc_sdk_examples_tools/config/sdks.yaml [hunk 6]
```diff
@@ -122,6 +122,7 @@ Go:
           route-53-domains: route53domains
           route53-recovery-cluster: route53recoverycluster
           s3-control: s3control
+          s3-directory-buckets: s3
```
[the remark this answers is not in the document]
Comment (Laren-AWS) [reply aws_doc_sdk_examples_tools/config/sdks.yaml:125]: Good catch! I added it.

File: aws_doc_sdk_examples_tools/config/services.yaml (hunk 5 above)
Comment (Laren-AWS) [reply aws_doc_sdk_examples_tools/config/services.yaml:2787]: Done

File: aws_doc_sdk_examples_tools/config/services.yaml (hunk 1 above)
Comment (Laren-AWS) [reply aws_doc_sdk_examples_tools/config/services.yaml:2779]: Updated

File: aws_doc_sdk_examples_tools/config/services.yaml (hunk 3 above)
Comment (Laren-AWS) [reply aws_doc_sdk_examples_tools/config/services.yaml:2785]: Done

File: aws_doc_sdk_examples_tools/config/services.yaml (hunk 2 above)
Comment (Laren-AWS) [reply aws_doc_sdk_examples_tools/config/services.yaml:2782]: Done

**Review by cpyle0819 — APPROVED** [review]

**Pull request merged**